bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -137,21 +148,10 @@ void ClientsManager::deleteOldestReply(NodeIdType clientId) {
   ReqId earliestReplyId = 0;
   const uint16_t clientIdx = clientIdToIndex_.at(clientId);
   auto& repliesInfo = indexToClientInfo_.at(clientIdx).repliesInfo;
-  for (const auto& reply : repliesInfo) {
-    if (earliestTime > reply.second) {
-      earliestReplyId = reply.first;
-      earliestTime = reply.second;
-    }
-  }
-  if (earliestReplyId)
-    repliesInfo.erase(earliestReplyId);
-  else if (!repliesInfo.empty()) {
-    // Delete reply arrived through ST
-    auto const& reply = repliesInfo.cbegin();
-    earliestReplyId = reply->first;
-    earliestTime = reply->second;
-    repliesInfo.erase(reply);
-  }
+  // Since seqnum is always growing we can be sure that the first element on the map is the oldest reply so we want to
+  // remove this reply.
+  earliestReplyId = repliesInfo.cbegin()->first;
```

Comment:
Verify that repliesInfo is not empty first.

Reply:
Fixed, thanks.